packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -174,6 +174,13 @@ abstract class SliverChildDelegate {
   /// away.
   bool shouldRebuild(covariant SliverChildDelegate oldDelegate);
 
+  /// Find index of child element with associated key.
+  ///
+  /// This will be called during [performRebuild] in [SliverMultiBoxAdaptorElement]
+  /// to check if a child has moved to a different position. It should return the
+  /// index of the child element with associated key, -1 if not found.
+  int findIndexByKey(Key key) => -1;
```

Comment:
returning `null` is probably more idiomatic than -1

Reply:
I usually expect -1 if not found.
https://api.dartlang.org/stable/2.2.0/dart-core/List/indexOf.html
Is there a reason why null is more preferable here?

Reply:
We might want to make -1 a valid index in some list view APIs.

Reply:
fixed